releases/stable-branches-and-release-cycle.md

```diff
@@ -0,0 +1,79 @@
+# Producing and supporting stable releases
+
+## Summary
+
+Today the Submariner project releases one minor version a month without capacity
+for stable branches or bugfix releases. In the following proposal we define a plan
+to make this possible starting with 0.8.
+
+## Proposal
+
+### Alignment of branch -> image releases
+
+We produce currently `<image>:devel` for every master push, along with `<image>:x.y.z` for
+the releases.
+
+Changing the `master` branch name to `devel` would make this homogeneous so our image releases
+are simply `<image>:<branch-or-tag>`.
+
+### Creating one stable branch for every minor release
+
+Approaching the release of every minor or major version a new stable branch will be forked
+off `devel` branch in the form of `release-x.y` for every existing repository.
+
+Any changes related to pinning shipyard base image will be done in that branch.
+
+At that point in time, any patches which belong to the release will need to be merged in master
+and backported to `release-x.y`. For this reason, the later we branch, the less extra work
+will be necessary to produce a stable release.  
+
+### Maintaining a stable branch
```

Comment:
I think we're missing here that we're only expecting backports of bugs not features (except exceptions).

Reply:
+1 hotfixes or critical bugs

Reply:
At the start probably it has to be more than critical bugs,

normally in other software, first it's general bugs, for some time, and then at some point, it becomes only hotfixes and critical bugs.

Reply:
I think it's best to explicitly specify the policy then, for instance we can say the latest stable gets all butgfixes and the stable n-1 gets just critical fixes.

Reply:
Agreed, we also define (or point to our definition elsewhere) of what constitutes a critical bugfix. It might be worth mentioning here too what we want to do regarding security issues; for example, we might want to allow no-change bumps to rebuilds container images to get fixed dependencies in our container images.

Reply:
What do you mean by "allow no-change bumps to rebuilds container images to get fixed dependencies in our container images."

Reply:
I imagine there will be circumstances where one of the packages we use in our images, or even the base image itself, needs to be upgraded because of a security issue or critical bug. When that happens, we’ll want to rebuild our images and release them, with a new version number, even if there are no changes in our code — so the code for, say, 0.8.2 and 0.8.3 would be identical, but the container images would be different and include important fixes.

Reply:
Ah, makes sense. Thanks for the clarification.